Documentation/MigrationGuide.md

```diff
@@ -1,15 +1,116 @@
-# Master
+# SwiftGen 6.0 Migration Guide
+
+If you're migrating from SwiftGen 5.x to SwiftGen 6.0, there might be some migration steps you'll need to use.
+
+Below is a list of pointers to help you migrate to the new SwiftGen 6.0
+
+## `storyboards` command has been renamed
+
+The `storyboards` subcommand has been renamed `ib` (for Interface Builder).
+You should replace invocations in your config files with the new `ib` command name, or command line invocations of `swiftgen storyboards …` with `swiftgen ib …`.
+
+❗️ See below: the bundled `storyboards` templates have been split up into one for scenes and one for segues.
+
+## Bundled Templates have been renamed & cleaned
+
+We've (finally) removed the bundled `swift2` templates. These have been deprecated for a while and were not being tested, so we're dropping them with this release. We assume most of you are already developing in a newer version of Swift (3 or 4).
+
+The biggest change is that the `storyboards` (`ib`) templates have been split up into separate templates, grouping functionality into a specific template. There's now a template for your scene information `scenes-swift3/4`, and a template for your segue information `segues-swift3/4`. We've split these up in preparation for more upcoming functionality, such as accessibility labels. To help you generating multiple outputs, see [below](#commands-can-have-multiple-outputs).
+
+A few minor template functionality changes have been made, mostly cleaning up some old code and old behaviours, you can read more about it in the [templates Migration Guide](Templates/MigrationGuide.md#swiftgen-60-migration-guide).
+
+## Commands can have multiple outputs
+
+This is only available for users with a configuration file, not via command line invocation. For each command you can now have a list of `outputs`, each output with a template name (`templateName`) or path (`templatePath`), and a list of parameters (`params`). For example:
+
+```yaml
+ib:
+  inputs: dir/to/search/for/storyboards
+  outputs:
+    - templateName: scenes-swift4
+      output: Storyboard Scenes.swift
+    - templateName: segues-swift4
+      output: Storyboard Segues.swift
+```
+
+As an extra advantage, the resources for that command will only be parsed once, giving you a nice performance boost.
+
+You'll notice that the configuration structure has changed a little bit, see below for more information.
+
+## Configuration changes
+
+Now that each command can have multiple outputs, we're deprecating the following configuration settings.
+
+- `output` has been renamed `outputs`, which now accepts one or more items.
+- `templateName`, `templatePath` and `params` have been moved down to the `outputs` level, so that each output can have a template name (or path) and parameters.
```

Comment:
- `templateName`, `templatePath`, `output` and `params` have been moved down to the `outputs` level, so that each output can have a template name (or path), a output file path, and optional parameters.